### PR TITLE
Make loader ESM by default, with UMD fallback

### PR DESCRIPTION
**Pull request by dcodeIO**
The loader cannot be easily used on the web currently without either bundling or exports-shimming it, so this PR refactors it to an ES module by default with an UMD fallback. The package is set up in a way that either `require`ing (from CommonJS) or `import`ing (from ESM) the loader should work in recent node.js, and where that's not applicable for some reason one may `require("@assemblyscript/loader/umd")` as a fallback.

Should not break backwards compatibility under recent node.js, but will break usage under older node.js and on the web (to the better, esp. on the web), e.g. will require updating the editor on our website by removing the exports-shim.

Unlike https://github.com/AssemblyScript/assemblyscript/pull/1313 this does not do any silly `.mjs` or `.cjs` stuff (turned out to break CDNs, i.e. not understanding what a .cjs's content type is) and does not refactor anything else but the loader, which is the most important for now.

- [x] I've read the contributing guidelines

### CODE REVIEW

File: lib/loader/index.js [hunk 1]
```diff
@@ -420,4 +412,9 @@ function demangle(exports, extendedExports = {}) {
   return extendedExports;
 }
 
-exports.demangle = demangle;
+export default {
+  instantiate,
+  instantiateSync,
+  instantiateStreaming,
+  demangle
+};
```
Comment (MaxGraey) [inline lib/loader/index.js:420]: Hmm but all this methods already marked as exported. Is not this redundant?

Comment (dcodeIO) [reply lib/loader/index.js:420]: The `default` export is what becomes the transpiled exports of the CommonJS module, i.e.

```js
const loader = require("loader")
// ===
import loader from "loader"; // default export
```

Comment (MaxGraey) [reply lib/loader/index.js:420]: I see, thanks!

**Review by MaxGraey — APPROVED** [review]

**Comment by dcodeIO** [comment]
Erm, when I put the `exports` key into `package.json`, a `require("assemblyscript")` inside of the package finds the package, otherwise it doesn't. That's unexpected.

**Comment by dcodeIO** [comment]
Seems that'll break on node 12 LTS. Ouch.

**Comment by dcodeIO** [comment]
Ok, turns out we can't know for sure anymore what `require("assemblyscript")` requires in newer node versions. A little unfortunate, but given that this is all set up and will probably change anyway once there's a new LTS, it should be fine to remove the respective checks for the time being so we can move on.

**Pull request merged**